bash/library/bench-base

```diff
@@ -121,3 +121,91 @@ function validate_clocksource() {
         return 1
     fi
 }
+
+function disable_balance_v1()
+{
+    local cpulist="$1"; shift
+	local cpu=""
+	local file=
+	local flags_cur=
+	for cpu in `echo ${cpulist} | sed -e 's/,/ /g'`; do
+		for file in $(find /proc/sys/kernel/sched_domain/cpu$cpu -name flags -print); do
+			flags_cur=$(cat $file)
+			flags_cur=$((flags_cur & 0xfffe))
+			echo $flags_cur > $file
+		done
+	done
+}
+
+function enable_balance_v1()
+{
+    local cpulist="$1"; shift
+	local cpu=""
+	local file=
+	local flags_cur=
+	for cpu in `echo ${cpulist} | sed -e 's/,/ /g'`; do
+		for file in $(find /proc/sys/kernel/sched_domain/cpu$cpu -name flags -print); do
+			flags_cur=$(cat $file)
+			flags_cur=$((flags_cur | 0x1))
+			echo $flags_cur > $file
+		done
+	done
+}
+
+function enable_balance_v2()
+{
+    echo member > /sys/fs/cgroup/user.slice/cpuset.cpus.partition
+    echo -cpuset > /sys/fs/cgroup/cgroup.subtree_control
+}
+
+function disable_balance_v2()
+{
+    local cpulist="$1"; shift
+    echo +cpuset > /sys/fs/cgroup/cgroup.subtree_control
+    echo $cpulist > /sys/fs/cgroup/user.slice/cpuset.cpus
+    echo isolated > /sys/fs/cgroup/user.slice/cpuset.cpus.partition
+}
```

Comment:
So I know very little about this, but are sure this is all that is needed?  Doesn't something need to put the tasks of interest into the cgroup in question (user.slice)?  If so, would we really want to use a generic cgroup like user.slice of should we be creating our own for more targeted (and isolated behavior since nobody else will be using it)?

@atheurer any input?

Reply:
Well, I think there is always potential for something else to be already using a "well-known" cgroup like user.slice (or added to it later).  If we reuse a "well-known" cgroup then we may inadvertently create a situation where we include other tasks in the cpuset where we are isolating things -- and that would be bad.  If we create a custom cgroup with an obscure enough name then we should avoid any such possible conflict down the road...I think.

Reply:
custom cgroups to be created

Reply:
```
# cat disable_balance
#!/bin/bash
set -x
echo +cpuset > /sys/fs/cgroup/cgroup.subtree_control
mkdir -p /sys/fs/cgroup/crucible_no_balance
echo 2-31 > /sys/fs/cgroup/crucible_no_balance/cpuset.cpus
echo isolated > /sys/fs/cgroup/crucible_no_balance/cpuset.cpus.partition
set +x

# cat enable_balance
#!/bin/bash
set -x
echo member > /sys/fs/cgroup/crucible_no_balance/cpuset.cpus.partition
echo -cpuset > /sys/fs/cgroup/cgroup.subtree_control
rmdir /sys/fs/cgroup/crucible_no_balance
set +x
```

disable:
```
# ./disable_balance
+ echo +cpuset
+ mkdir -p /sys/fs/cgroup/crucible_no_balance
+ echo 2-31
+ echo isolated
+ set +x
# dmesg | tail -n10 | grep "domain span"
[525501.031209] root domain span: 0-1
```

enable:
```
# ./enable_balance
+ echo member
+ echo -cpuset
+ rmdir /sys/fs/cgroup/crucible_no_balance
+ set +x
# dmesg | tail -n10 | grep "domain span"
[525556.549206] root domain span: 0-31
```

@k-rister Does it address your point ?

Reply:
I think so.  I will admit that some of the details are slightly confusing to me, but I'm pretty sure that is all cgroup/cpuset related -- it seems like `-cpuset` (enable?) and `+cpuset` (disable?) are the opposite of what I would have expected?

Reply:
With +cpuset you activate the cpuset controller for constraining the cpu placement of tasks to the resources specified in the cpuset interface files inside a cgroup. To disable load balance you write "isolated" to cpuset.cpus.partition of a cgroup containing the cpuset.cpus, the range of cpus that load balance will be disabled.

Reply:
ack, makes sense